libcaf_core/caf/actor_system_config.hpp

```diff
@@ -5,6 +5,7 @@
 #pragma once
 
 #include "caf/actor_factory.hpp"
+#include "caf/actor_from_state.hpp"
```

Comment:
Is there a way to not include that header here? It will pull in things like `actor_system.hpp`, which we should avoid including here if possible.

Reply:
I can refactor the code not to use `actor_from_state` here, but including `actor_factory.hpp` will still pull the `actor_from_state.hpp` header file transitively.